airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx

```diff
@@ -38,13 +40,16 @@ export type AuthSendEmailVerification = () => Promise<void>;
 export type AuthVerifyEmail = (code: string) => Promise<void>;
 export type AuthLogout = () => Promise<void>;
 
+type OAuthLoginState = "waiting" | "loading" | "done";
+
 interface AuthContextApi {
   user: User | null;
   inited: boolean;
   emailVerified: boolean;
   isLoading: boolean;
   loggedOut: boolean;
   login: AuthLogin;
+  loginWithOAuth: (provider: OAuthProviders) => Observable<OAuthLoginState>;
```

Comment:
:information_source: This method returns an Obersvable, since we need to know in the UI, when the middle of the method starts (the user selected an account, and we start loading the user), to show the loading spinner. Thus we can't express that with a Promise.